ui/src/models/workflows.ts

```diff
@@ -539,6 +539,15 @@ export interface Workflow {
 
 export const execSpec = (w: Workflow) => Object.assign({}, w.status.storedWorkflowTemplateSpec, w.spec);
 
+// The label may not have been updated on time but usually this indicates that they are already archived.
+export function isArchivedWorkflow(wf: Workflow): boolean {
+    return (
+        wf.metadata.labels &&
+        (wf.metadata.labels['workflows.argoproj.io/workflow-archiving-status'] === 'Archived' ||
```

Comment:
Yes, however, the result of a `workflowServer.List` call will only contain either the live wf or the archived wf (to avoid duplicating the entry in the list view). If the live wf is used, this label will be "Archived". If the one in the database is used, then this label will be "Pending".

Reply:
Improved and corrected this in https://github.com/argoproj/argo-workflows/pull/11336